ksql-examples/src/main/java/io/confluent/ksql/datagen/DataGen.java

```diff
@@ -166,6 +166,7 @@ private static void usage() {
     private final Supplier<InputStream> schemaFile;
     private final Format keyFormat;
     private final Format valueFormat;
+    private final Character valueDelimiter;
```

Comment:
```suggestion
    private final char valueDelimiter;
```

Would be good to not make `null` an issue by using an unboxed type and just setting it to the default ',' rather than `null`.

Reply:
I think it's ok for it to be null in the case the the value format is not delimited.

Reply:
The encode its optionality into the API, i.e. 

```suggestion
    private final Optional<Character> valueDelimiter;
```

Better still, introduce the `Delimiter` class we've discussed to make it even more type safe and have

```suggestion
    private final Optional<Delimiter> valueDelimiter;
```

Reply:
I agree that DataGen should probably be refactored at some point to use Optional and not null/not null to represent optional/mandatory args, but I've done it this way to fit in with the way DataGen currently works as I didn't consider it within the scope of this PR to refactor DataGen.